controllers/machinehealthcheck_targets.go

```diff
@@ -135,14 +135,18 @@ func (t *healthCheckTarget) needsRemediation(logger logr.Logger, timeoutForMachi
 		}
 		logger.V(3).Info("Using comparison time", "time", comparisonTime)
 
-		if comparisonTime.Add(timeoutForMachineToHaveNode).Before(now) {
+		if timeoutForMachineToHaveNode != time.Duration(0) && comparisonTime.Add(timeoutForMachineToHaveNode).Before(now) {
 			conditions.MarkFalse(t.Machine, clusterv1.MachineHealthCheckSuccededCondition, clusterv1.NodeStartupTimeoutReason, clusterv1.ConditionSeverityWarning, "Node failed to report startup in %s", timeoutForMachineToHaveNode.String())
 			logger.V(3).Info("Target is unhealthy: machine has no node", "duration", timeoutForMachineToHaveNode.String())
 			return true, time.Duration(0)
 		}
 
 		durationUnhealthy := now.Sub(comparisonTime)
 		nextCheck := timeoutForMachineToHaveNode - durationUnhealthy + time.Second
+		if timeoutForMachineToHaveNode == time.Duration(0) {
```

Comment:
same question about disable value here

Reply:
Could we use timeoutForMachineToHaveNode.IsZero() instead?